modules/navier_stokes/src/auxkernels/NSMachAux.C

```diff
@@ -28,20 +33,14 @@ NSMachAux::NSMachAux(const InputParameters & parameters) :
     _u_vel(coupledValue("u")),
     _v_vel(_mesh.dimension() >= 2 ? coupledValue("v") : _zero),
     _w_vel(_mesh.dimension() == 3 ? coupledValue("w") : _zero),
-    _temperature(coupledValue("temperature")),
-    _gamma(getParam<Real>("gamma")),
-    _R(getParam<Real>("R"))
+    _specific_volume(coupledValue("specific_volume")),
+    _internal_energy(coupledValue("internal_energy")),
+    _fp(getUserObject<IdealGasFluidProperties>("fluid_properties"))
 {
 }
 
 Real
 NSMachAux::computeValue()
 {
-  const RealVectorValue vel(_u_vel[_qp], _v_vel[_qp], _w_vel[_qp]);
-
-  // If temperature is negative, the solution is probably not going to
-  // be very good anyway... in that case we will just compute the Mach
-  // number based on |T| and avoid NaNs.  We don't check whether
-  // temperature is somehow exactly 0.0, though.
-  return vel.norm() / std::sqrt(_gamma * _R * std::abs(_temperature[_qp]));
+  return RealVectorValue(_u_vel[_qp], _v_vel[_qp], _w_vel[_qp]).norm() / _fp.c(_specific_volume[_qp], _internal_energy[_qp]);
```

Comment:
`c` is computed by the `FluidPropertiesMaterial`, so you can get it via `getMaterialProperty`.